### ISSUE TITLE
bad type narrowing with nested assert

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#423]
```lua
--- @return string?
local function bar() end

--- @param a? string
function foo(a)
  if not a then
    a = bar()
    assert(a)
  end

  --- @type string
  local _ = a -- incorrect error
end
```
